### PR TITLE
feat(upgrade): allow upgrade to develop

### PR DESCRIPTION
**Pull request by sinhaashish**
This PR allows the upgrade to develop branch and updates the destination version of upgrade to 2.2.0

### CODE REVIEW

**Comment by sinhaashish** [comment]
bors try

File: k8s/upgrade-job/src/helm/upgrade.rs [hunk 1]
```diff
@@ -137,13 +137,15 @@ impl HelmUpgradeBuilder {
...
+
         let from_version = upgrade::path::version_from_release_chart(chart)?;
         let upgrade_path_is_valid =
             upgrade::path::is_valid(chart_variant.clone(), &from_version, &to_version)?;
+        let upgrade_to_develop = upgrade::path::upgrade_to_develop(&from_version, &to_version)?;
         let already_upgraded = to_version.eq(&from_version);
 
         ensure!(
-            upgrade_path_is_valid || already_upgraded,
+            upgrade_path_is_valid || already_upgraded || upgrade_to_develop,
```
Comment (niladrih) [inline k8s/upgrade-job/src/helm/upgrade.rs:148]: 👍

File: k8s/upgrade-job/src/upgrade/path.rs [hunk 2]
```diff
@@ -43,6 +46,20 @@ pub(crate) fn is_valid(chart_variant: HelmChart, from: &Version, to: &Version) -
     }
 }
 
+/// Validates the upgrade to target as develop branch.
```
Comment (niladrih) [inline k8s/upgrade-job/src/upgrade/path.rs:49]:
```suggestion
/// Validates the upgrade path from the 'from' Version to a helm chart in the develop branch of this repo.
/// E2e tests upgrade from the latest released helm chart to the chart present at the HEAD commit of the develop 
/// branch at the time.
```

**Review by niladrih — COMMENTED** [review]
Left few comments..

**Comment by sinhaashish** [comment]
Closing this PR  as another PR https://github.com/openebs/mayastor-extensions/pull/229 is raised for the same

**Pull request closed without merging**